documentation/src/main/asciidoc/topics/ref_cache_store_sfs.adoc

```diff
@@ -19,9 +19,26 @@ defragmented.
 
 .Declarative configuration
 
+* For embedded deployments, {brandname} stores data to the current working
+directory. To store data in a different location, specify the absolute path to
+that directory as the value, as in the following example:
+
+[source,xml,options="nowrap",subs=attributes+]
+----
+<persistence>
+   <file-store path="/path/to/myDataStore" max-entries="5000"/>
+</persistence>
+----
+
+* For server deployments, {brandname} stores data in `{server_home}/data`. You
+should not specify a `path` for Single File cache stores with {brandname} 
+server.
+
 [source,xml,options="nowrap",subs=attributes+]
 ----
-include::config_examples/persistence_file_store.xml[]
+<persistence>
+   <file-store max-entries="5000"/>
+</persistence>
```

Comment:
@ryanemerson I'm curious about the programmatic configuration. docs provide this example:

```
ConfigurationBuilder b = new ConfigurationBuilder();
b.persistence()
    .addSingleFileStore()
    .location("myDataStore")
    .maxEntries(5000);
``` 

this is for library right? is there a way to configure persistence via Hot Rod other than using XMLStringConfiguration to pass a cache definition that includes file-store?

Reply:
Yep this is for library. There's no other way to configure persistence via Hot Rod as there is no other way to configure caches. Persistence configuration is just a sub-module of the overall cache configuration which doesn't get any special treatment.

Reply:
FYI that Tristan added this to the Hot Rod client props recently: https://github.com/infinispan/infinispan/blob/1ffe539cf99996a06250d01915a43296474ab357/client/hotrod-client/src/main/java/org/infinispan/client/hotrod/configuration/package-info.java#L413

Reply:
actually here's RemoteCacheConfigurationBuilder.java https://github.com/infinispan/infinispan/commit/4ecc4992323ee5612e4270de8166f369e7bb0634#diff-f98cef32f0dccd584e5609d2e33b48b9R68

Reply:
Yep,  that PR was badly timed with my comment :smile:

Reply:
ha. I wasn't sure if you'd seen that or not. it actually helps me solve a problem because docs often show programmatic examples that only apply to embedded but we don't always make that clear.